src/main/java/me/snowdrop/licenses/xml/DependencyElement.java

```diff
@@ -42,18 +44,20 @@
     private String version;
 
     private Set<LicenseElement> licenses;
+    
+    private Artifact artifact;
 
     public DependencyElement() {
     }
 
-    public DependencyElement(MavenProject mavenProject) {
-        this.groupId = mavenProject.getGroupId();
-        this.artifactId = mavenProject.getArtifactId();
-        this.version = mavenProject.getVersion();
-        this.licenses = mavenProject.getLicenses()
-                .stream()
-                .map(LicenseElement::new)
-                .collect(Collectors.toSet());
+
+
+    public DependencyElement(Artifact artifact) {
+        this.groupId = artifact.getGroupId();
+        this.artifactId = artifact.getArtifactId();
+        this.version = artifact.getVersion();
+
+        this.artifact = artifact;
```

Comment:
I don't think we should keep artifact in this object. We can always recreate it with groupId, artifactId and version when needed. Also most of the time it will be null because other constructors to sent it.